### ISSUE TITLE
[cli] Improve project-type provider argument validation

### ISSUE THREAD

**Issue by markusheiliger** [microsoft/TeamCloud#85]
**Describe the bug**
The command failed with an unexpected error. Here is the traceback:
```
not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 572, in execute
    parsed_args = self.parser.parse_args(args)
  File "/opt/az/lib/python3.6/site-packages/knack/parser.py", line 260, in parse_args
    return super(CLICommandParser, self).parse_args(args)
  File "/opt/az/lib/python3.6/argparse.py", line 1730, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/opt/az/lib/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/opt/az/lib/python3.6/argparse.py", line 1950, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/opt/az/lib/python3.6/argparse.py", line 1927, in consume_positionals
    take_action(action, args)
  File "/opt/az/lib/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/opt/az/lib/python3.6/argparse.py", line 1133, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/opt/az/lib/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/opt/az/lib/python3.6/argparse.py", line 1950, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/opt/az/lib/python3.6/argparse.py", line 1927, in consume_positionals
    take_action(action, args)
  File "/opt/az/lib/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/opt/az/lib/python3.6/argparse.py", line 1133, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/opt/az/lib/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/opt/az/lib/python3.6/argparse.py", line 1950, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/opt/az/lib/python3.6/argparse.py", line 1927, in consume_positionals
    take_action(action, args)
  File "/opt/az/lib/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/opt/az/lib/python3.6/argparse.py", line 1133, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/opt/az/lib/python3.6/argparse.py", line 1762, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/opt/az/lib/python3.6/argparse.py", line 1968, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/opt/az/lib/python3.6/argparse.py", line 1908, in consume_optional
    take_action(action, args, option_string)
  File "/opt/az/lib/python3.6/argparse.py", line 1836, in take_action
    action(self, namespace, argument_values, option_string)
  File "/home/markushe/.azure/cliextensions/tc/azext_tc/_actions.py", line 37, in __call__
    key, value = item.split('=', 1)
ValueError: not enough values to unpack (expected 2, got 1)
```
**To Reproduce**
 az tc project-type create --name default --subscriptions [GUID] --provider azure.appinsights azure.devops azure.devtestlab --location EastUS --resource-group-name-prefix tcmarkus_ --default

**Comment by colbylwilliams** [comment]
Multiple `--provider` arguments are required to add more than one provider.  The value for each `--provider` is in the form of id then (optional) properties:

```sh
az tc project-type create -l eastus -n default \
--subscriptions GUID1 GUID2 GUID3 \
--provider azure.appinsights prop1=prop1 prop2=prop2 \
--provider azure.devtestlabs prop1=val1 depends_on=azure.appinsights
```
or in a single line:
```sh
az tc project-type create -l eastus -n default --subscriptions GUID1 GUID2 GUID3 --provider azure.appinsights prop1=prop1 prop2=prop2 --provider azure.devtestlabs prop1=val1 depends_on=azure.appinsights
```

Also use the "special" property key `depends_on` with the id of a provider to declare dependencies (in the example above).

Updating the issue to improve the argument validation and error message.

@markusheiliger

**Comment by colbylwilliams** [comment]
Fixed in [v0.3.0](https://www.myget.org/feed/teamcloud-dev/package/pythonwhl/tc).  Please [update](https://github.com/microsoft/TeamCloud/blob/master/docs/CLI.md#update).
